sql/src/provider.rs

```diff
@@ -218,27 +223,15 @@ impl<'a, P: MetaProvider> ContextProvider for ContextProviderAdapter<'a, P> {
         name: TableReference,
     ) -> std::result::Result<Arc<(dyn TableSource + 'static)>, DataFusionError> {
         // Find in local cache
-        if let Some(p) = self.table_cache.borrow().get(name) {
-            return Ok(p);
+        if let Some(table_ref) = self.table_cache.borrow().get(name) {
+            return Ok(self.table_source(table_ref));
         }
 
         // Find in meta provider
         match self.meta_provider.table(name) {
             Ok(Some(table)) => {
-                let table_adapter = Arc::new(TableProviderAdapter::new(
```

Comment:
We shouldn't cache `TableProviderAdapter` directly, since it contains request_id, which is different between different request.

If we cache them, request_id will be same across different requests.

Reply:
I find that the request id has been removed from the `TableProviderAdapter`, and if so, the `TableProviderAdapter` seems able to be cached directly?